pkg/docker/docker.go

```diff
@@ -581,20 +518,56 @@ func (d *stiDocker) CheckImage(name string) (*api.Image, error) {
 	return nil, nil
 }
 
+func base64EncodeAuth(auth dockertypes.AuthConfig) (string, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(auth); err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(buf.Bytes()), nil
+}
+
 // PullImage pulls an image into the local registry
 func (d *stiDocker) PullImage(name string) (*api.Image, error) {
 	name = getImageName(name)
-	err := d.kubeDockerClient.PullImage(name, d.pullAuth, dockertypes.ImagePullOptions{})
+
+	// RegistryAuth is the base64 encoded credentials for the registry
+	base64Auth, err := base64EncodeAuth(d.pullAuth)
 	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	resp, err := d.InspectImage(name)
+	opts := dockertypes.ImagePullOptions{RegistryAuth: base64Auth}
+	ctx := context.Background() // no timeout
+	resp, err := d.client.ImagePull(ctx, name, opts)
 	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	if resp != nil {
+	defer resp.Close()
+	reporter := newProgressReporter(name, func() {})
+	reporter.start()
+	defer reporter.stop()
+	decoder := json.NewDecoder(resp)
+	for {
+		var msg dockermessage.JSONMessage
+		err = decoder.Decode(&msg)
+		if err == io.EOF {
```

Comment:
I see - gotcha.  

Long winded aside:  Conversely, FWIW, we were building up a list of issues of similar feel and severity we had found in the k8s client during the original port.  I don't recall this one in particular - probably just slipped through.  But before we got too far down the path of working with the folks upstream to start addressing them, the winds around major rework in that space came, and then ultimately the remove effort you are now undertaking.

Brief summary of the aside: I'm not surprised such shortcomings were not previously addressed :-)

Circling back up:  are you amenable to improving this code borrowed from the k8s client ?